.github/workflows/run-googletest.yml

```diff
@@ -0,0 +1,33 @@
+# Run C++ tests using GoogleTest
+name: run-googletest
+# on specifies the build triggers. See more info at https://docs.github.com/en/actions/learn-github-actions/events-that-trigger-workflows
+on:
+# The default build trigger is to run the action on every push and pull request, for any branch
+  push:
+  pull_request:
+  # To run the default repository branch weekly on sunday, uncomment the following 2 lines
+  #schedule:
+    #- cron: '0 0 * * 0'
+jobs:
+  build:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, macos-latest, windows-latest]
+
+    steps:
+    - uses: actions/checkout@v2
+    - uses: seanmiddleditch/gha-setup-ninja@master # https://github.com/seanmiddleditch/gha-setup-ninja
+
+    - name: Configure
+      run: |
+        cmake -S . -B build -G Ninja
+        
+    - name: Build
+      run: |  
+        cmake --build build
+        
+    - name: Test
+      run: |
+        ctest --test-dir build
```

Comment:
Will the job fail if the Test step have failing tests? We may also want to consider saving artifacts (perhaps conditionally if a job fails), if there is something useful to save (e.g., a detailed summary of which tests failed and why).

Reply:
Yes, the job will fail if there are failing tests. Good idea for saving artifacts. I added this suggestion as a new task in issue #7.